aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -374,19 +402,47 @@ EOF
     $fh->close();
 }
 
-# Prints an IP address in hexadecimal.
-sub hexip
+
+# Write the Grub2 configuration file.
+# TODO: handle append options?
+sub write_grub2_config
 {
-    my $ip = shift || "";
+    my $cfg = shift;
+    my $t = $cfg->getElement (PXEROOT)->getTree;
+    my $fh = CAF::FileWriter->open (filepath ($cfg, PXE_VARIANT_GRUB2),
+                                    log => $this_app, mode => 0644);
 
-    return sprintf ("%02X%02X%02X%02X", $1, $2, $3, $4) if ($ip =~ m/^(\d+)\.(\d+)\.(\d+)\.(\d+)$/);
+    my $linux_cmd = $this_app->option(GRUB2_EFI_LINUX_CMD);
+    unless ( $linux_cmd ) {
+        $this_app->error("AII option ".GRUB2_EFI_LINUX_CMD." undefined");
```

Comment:
if possible, check this earlier, so it is reported once (and fails once). the configuration should be modified only once (but the defaults are probably ok to make this backwards compatible).